client/src/worldgen.rs

```diff
@@ -439,17 +445,26 @@ struct EnviroFactors {
     rainfall: i64,
     slopeiness: i64,
     blockiness: i64,
+    flatness: i64,
 }
 impl EnviroFactors {
     fn varied_from(parent: Self, spice: u64) -> Self {
         let mut rng = rand_pcg::Pcg64Mcg::seed_from_u64(spice);
         let plus_or_minus_one = Uniform::new_inclusive(-1, 1);
+        let flatness = (parent.flatness + rng.sample(&plus_or_minus_one))
+            .max(1)
+            .min(40);
         let slopeiness = parent.slopeiness + rng.sample(&plus_or_minus_one);
         Self {
             slopeiness,
+            flatness,
             max_elevation: parent.max_elevation
-                + ((3 - parent.slopeiness.rem_euclid(7)) + (3 - slopeiness.rem_euclid(7)))
-                + rng.sample(&plus_or_minus_one),
+                + ((((3 - parent.slopeiness.rem_euclid(7)) as f64)
+                    * (1.0 - (((parent.flatness as f64) - 20.0) / 10.0).tanh())
```

Comment:
`flatness - 20` has a range of -19 to 20 inclusive. That's asymmetrical, and I think therefore unsound; is it what you intended?

Reply:
It does not need to be symmetric. The 20.0 can be adjusted artistically. However, I might as well make it symmetric for neatness's sake.

Reply:
On review, the randomized sign at the end probably means any asymmetry here gets worked out in the end, but I still suspect it wasn't intended.

Reply:
Besides, 1 - tanh(x) is always more than zero.